### ISSUE TITLE
 Document new debug helper methods

### ISSUE THREAD

**Issue by TarantoolBot** [tarantool/doc#793]
There are two more helpers in debug module `sourcefile()` and
`sourcedir()`. This two helpers returns relative paths to source file
and directory respectively.

There are also `debug.__file__` and `debug.__dir__` which are just handy
ways to call `sourcefile()` and `sourcedir()`.

It should be mentioned that it is only possible to determine real path
to the file, if the function was defined in a some lua file.
`loadstring` may be exception here, since lua will store whole string
in debug info.

There is also a possible pitfall because of tail call optimization, so
if the function `fn` is defined inside `myapp/init.lua` like this:

```lua
function fn()
    return debug.sourcefile()
end
```

It would not be possible to determine file path correctly because that
function would not appear in a stacktrace. Even worse, it may return
wrong result, i.e. actual path where "parent" function was defined.

To force the interpreter to avoit this kind of optimizations you may use
parenthesis like this:

```lua
function fn()
    return (debug.sourcefile())
end
```

Also both `sourcefile` and `sourcedir` functions have optional level
argument, which allows set the level of the call stack function should
examine for source path. By default, 2 is used.
Requested by @RunsFor in https://github.com/tarantool/tarantool/commit/70b8583ca4ac4162b675b2a7fed074daa43ff5e3.

**Comment by olegrok** [comment]
Backported to 1.10 https://github.com/tarantool/tarantool/commit/6a7913d5448457d154ac4e5b358cd3691788bec0

@lenkis change tag from 2.1 to 1.10, please

**Comment by pgulutzan** [comment]
The fix is in 2.1, as @lenkis specified.
